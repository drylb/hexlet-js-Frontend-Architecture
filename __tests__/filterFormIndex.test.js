/* eslint-disable no-undef */
import '@testing-library/jest-dom';
import fs from 'fs';
import path from 'path';
import prettier from 'prettier';
import testingLibraryDom from '@testing-library/dom';
import testingLibraryUserEvent from '@testing-library/user-event';

import run from '../src/filterForm';

const { screen } = testingLibraryDom;
const userEvent = testingLibraryUserEvent.default;

const options = {
  parser: 'html',
  htmlWhitespaceSensitivity: 'ignore',
  tabWidth: 4,
};

const getTree = (el) => prettier.format(el.innerHTML, options);

beforeEach(() => {
  const initHtml = fs.readFileSync(path.join('__fixtures__', 'filterFormIndex.html')).toString();
  document.body.innerHTML = initHtml;
  run();
});

test('working process', async () => {
  const result = document.querySelector('.result');
  expect(getTree(result)).toMatchSnapshot();

  const frequencyMin = await screen.findByLabelText('Frequency Min');
  userEvent.type(frequencyMin, '3');
  expect(getTree(result)).toMatchSnapshot();

  userEvent.clear(frequencyMin);
  expect(getTree(result)).toMatchSnapshot();

  userEvent.type(frequencyMin, '4');
  expect(getTree(result)).toMatchSnapshot();

  userEvent.clear(frequencyMin);
  userEvent.type(frequencyMin, '1');
  expect(getTree(result)).toMatchSnapshot();

  const frequencyMax = await screen.findByLabelText('Frequency Max');
  userEvent.type(frequencyMax, '2');
  expect(getTree(result)).toMatchSnapshot();
});

test('initial state', async () => {
  const result = document.querySelector('.result');
  expect(getTree(result)).toMatchSnapshot();
});
