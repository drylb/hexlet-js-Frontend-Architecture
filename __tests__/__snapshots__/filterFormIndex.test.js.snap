// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial state 1`] = `
"<ul>
    <li>v1</li>
    <li>d3</li>
    <li>d2</li>
</ul>
"
`;

exports[`working process 1`] = `
"<ul>
    <li>v1</li>
    <li>d3</li>
    <li>d2</li>
</ul>
"
`;

exports[`working process 2`] = `
"<ul>
    <li>d3</li>
</ul>
"
`;

exports[`working process 3`] = `
"<ul>
    <li>v1</li>
    <li>d3</li>
    <li>d2</li>
</ul>
"
`;

exports[`working process 4`] = `""`;

exports[`working process 5`] = `
"<ul>
    <li>v1</li>
    <li>d3</li>
    <li>d2</li>
</ul>
"
`;

exports[`working process 6`] = `
"<ul>
    <li>v1</li>
</ul>
"
`;
